Chapter-3-Exercises/jjhampton-ch3-bean-counting.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+/* -------------------------------
+Write a function countBs that takes a string as its only argument and returns a number that indicates how many uppercase “B” characters are in the string.
+
+Next, write a function called countChar that behaves like countBs, except it takes a second argument that indicates the character that is to be counted (rather than counting only uppercase “B” characters). Rewrite countBs to make use of this new function.
+-------------------------------*/
+
+function countBs(stringToCheck, charToCount) {
+    const stringLength = stringToCheck.length; // cache string length to prevent repeated lookups during each for loop iteration
+    let counter = 0;
+    for (let i = 0; i < stringLength; i++) {
+        if (stringToCheck.charAt(i) === charToCount) 
+            counter++
+    }
+    return counter;
+}
+
```

Comment:
Yeah - I haven't been using `use strict` too much lately, but probably should more often.  I've been using Typescript at work and AFAIK it actually adds that to its output JS in the newer versions.  In order to run my code locally on Node.js, I had to add `use strict` to use the ES2015 `let` and `const` keywords - not transpiling anything from ES6 > ES5.

Yeah, I saved the value of `stringToCheck.length` to a variable, so the program only has to look it up one time, as a performance advantage.  Otherwise, if it's in the for-loop as `for (let i = 0; i < stringToCheck.length; i++)`,  the for-loop will have to query the string for the value every iteration of the loop.

I don't think it really makes much of a difference on a small program like this, but I've tried to get into the habit of cacheing values into variables that will be looped over so the program only has to make the lookup once.